### ISSUE TITLE
Support background colors for table

### ISSUE THREAD

**Issue by kartikmandhan** [sjdemartini/mui-tiptap#253]
### Is your feature request related to a problem? Please describe.
Currently table, has only support for toggling header cell, can we have support to provide custom background color for a cell in the table, as part of tableBubbleMenu

**Comment by kartikmandhan** [comment]
Will this feature be supported in future releases? Or can you give me a starting point, where I should start to build this, and then raise the MR for it. As I tried adding background-color to TableCell component from the browser-developer tools as well, but it doesn't work for me.

**Comment by sjdemartini** [comment]
> Will this feature be supported in future releases? Or can you give me a starting point, where I should start to build this, and then raise the MR for it. As I tried adding background-color to TableCell component from the browser-developer tools as well, but it doesn't work for me.

I have no plans to implement this currently. In terms of where to start building this, I would recommend taking a look at the Tiptap docs on custom extensions https://tiptap.dev/docs/editor/extensions/custom-extensions. You will likely need to extend the TableCell extension, add a property to it, and use the property to render a color.

**Comment by kartikmandhan** [comment]
I just want the support for it, for example if an existing html has a background-color property that should render inside the editor.
`<th style="background-color: red;> head</th>` the css doesn't apply to this cell, will this part also need modification in the TableCell component of tiptap library or some modification in the mui-tiptap is sufficient for the usecase?
Please let me know the next steps to investigate @sjdemartini

**Comment by kartikmandhan** [comment]
The code below serves my usecase, but whether should I add this in mui-tiptap or not required is my ask, I can add the options in the Table bubble-menu as well if required, to support background-color for table cells.
```
import { TableCell } from "@tiptap/extension-table-cell";

const TableCellImproved = TableCell.extend({
  addAttributes() {
    return {
      ...this.parent?.(),
      backgroundColor: {
        default: null,
        renderHTML: (attributes) => {
          if (!attributes.backgroundColor) {
            return {};
          }

          return {
            style: `background-color: ${attributes.backgroundColor as string}`,
          };
        },
        parseHTML: (element) => {
          return element.style.backgroundColor.replace(/['"]+/g, "");
        },
      },
    };
  },
});

export default TableCellImproved;
```

**Comment by kartikmandhan** [comment]
Or In general, adding this snippet will support all the styles
```
import { TableCell } from "@tiptap/extension-table-cell";

const TableCellImproved = TableCell.extend({
  addAttributes() {
    return {
      ...this.parent?.(),
      style: {
        default: null,
        renderHTML: (attributes) => {
          if (!attributes.style) {
            return {};
          }

          return {
            style: attributes.style,
          };
        },
        parseHTML: (element) => {
          const style = element.getAttribute("style") || "";
          return style;
        },
      },
    };
  },
});

export default TableCellImproved;
```

**Comment by sjdemartini** [comment]
Thanks for sharing the examples in case others would like to see. I don't have plans to incorporate this into mui-tiptap at this time, and as you say it will require some relatively tricky modifications to the table bubble menu, but the beauty of tiptap is that you can modify as you need and bring your own extensions based on your own specific use case.

**Comment by kartikmandhan** [comment]
But we could at least add TableCellImproved via a PR, if you agree. That would enhance the rendering of HTML content by the editor.

**Comment by sjdemartini** [comment]
I think without additional functionality to utilize those new extension fields *within* mui-tiptap, and because it's so simple to define that sort of extension in a consuming project if someone wants it, there isn't much value in including that extension in mui-tiptap at this time.

**Comment by kartikmandhan** [comment]
Hi @sjdemartini , I would like to contribute via pr a custom extenstion for Table Cell Background color that can be included in EditorMenu. Attaching a screenshot of how this would work:
When a table cell is focused:
![Image](https://github.com/user-attachments/assets/6049ce50-1784-43cf-a1b6-30e50f254e4b)
When no table cell is focused it would be disabled:
![Image](https://github.com/user-attachments/assets/cadc93be-a912-4cae-8078-fc6775e9c40b)

Please let me know, if you think this is a valuable contribution

**Comment by kartikmandhan** [comment]
And just to avoid confusion I would like to suggest a different icon for Toggle Header Row:
![Image](https://github.com/user-attachments/assets/f57a362c-e132-44f2-b14f-47a88c8b5103)

Open for other icon suggestions as well.

**Comment by sjdemartini** [comment]
Glad you got this working in your own project! 

Hm, it seems confusing to me to have table cell background color editing happening in the main editor controls, while all other table controls are in the table bubble menu. I find the existing "toggle header row" icon more appropriate than the new one you suggest as well.

Given the complexity to get this right (UI/UX and otherwise), and this being a more niche custom thing, I think it may be preferable to still just add this functionality within your own app and not bake it into mui-tiptap.

**Comment by kartikmandhan** [comment]
Initially I added it inside the bubble menu only, but then when you click on the icon to make the color picker visible, the bubble menu disappears. So I made the color picker appear on hover action as well. But then it still was complex for the user to get it right color, as it out focused the bubble menu and made it disappear. Hence added near the main menu. Also if you think about this as a feature, Excel supports it as well right (that too in a main menu) I really think this would be a nice addition in the library. I would be glad to work on the icon part and ux of it to make it work for the project. Can you provide some suggestions on the Ux of it?

**Comment by sjdemartini** [comment]
Yeah, this is what I mean by the problem being tricky to solve given how mui-tiptap works today. I wouldn't expect for mui-tiptap to support much of the functionality of Excel, though I get the desire for this feature. I don't have a great suggestion for a solution here myself, hence thinking it makes more sense to table (no pun intended) adding this directly to mui-tiptap. Similar PR was abandoned earlier on adding different table style customization https://github.com/sjdemartini/mui-tiptap/pull/185
